### PR TITLE
Fix PodStore to wait for being initialized

### PR DESCRIPTION
**Pull request by wojtek-t**
Currently causing some e2e test failure (mostly restart nodes tests).

### CODE REVIEW

File: test/e2e/scalability/density.go [hunk 1]
```diff
@@ -226,8 +226,13 @@ func density30AddonResourceVerifier(numNodes int) map[string]framework.ResourceC
 
 func logPodStartupStatus(c clientset.Interface, expectedPods int, observedLabels map[string]string, period time.Duration, stopCh chan struct{}) {
 	label := labels.SelectorFromSet(labels.Set(observedLabels))
-	podStore := testutils.NewPodStore(c, metav1.NamespaceAll, label, fields.Everything())
+	podStore, err := testutils.NewPodStore(c, metav1.NamespaceAll, label, fields.Everything())
+	if err != nil {
+		framework.Logf("Couldn't initialize pod store: %v", err)
```
Comment (shyamjvs) [inline test/e2e/scalability/density.go]: Don't we want to framework.ExpectNoError() here?

File: test/e2e/framework/util.go [hunk 2]
```diff
@@ -3068,7 +3071,7 @@ func DeleteResourceAndWaitForGC(c clientset.Interface, kind schema.GroupKind, ns
 		return err
 	}
 
-	ps, err := podStoreForSelector(c, ns, selector)
+	ps, err := testutils.NewPodStore(c, ns, selector, fields.Everything())
```
Comment (MaciekPytel) [inline test/e2e/framework/util.go]: Is that actually equivalent to the previous version? In old code it's passing what looks like a specific selector and here you just pass fields.Everything?

Comment (MaciekPytel) [reply test/e2e/framework/util.go]: Ok, please ignore, just saw the method below.

File: test/e2e/scalability/density.go (hunk 1 above)
Comment (wojtek-t) [reply test/e2e/scalability/density.go]: done

**Comment by wojtek-t** [comment]
Once merged, we should cherrypick it to release branches to fix tests there.
@jpbetz @mbohlool - my recent fixes to tests exposed that problem

**Pull request merged**